client/packages/common/src/intl/utils/IntlUtils.ts

```diff
@@ -125,6 +126,15 @@ export const useIntlUtils = () => {
     return t(localeKey, Formatter.fromCamelCase(serverKey));
   };
 
+  const translateReportName = (code: string, name: string) => {
+    const localeKey = `report.${code}`
+    return isLocaleKey(localeKey) ? t(localeKey) : name;
+  }
+
+  const isLocaleKey = (key: string): key is LocaleKey => {
+    return localeKeySet.has(key);
+  };
```

Comment:
Check for localeKey type safety as opposed to casting.

I understand translation function has passable default, so maybe not necessary?